src/client.js

```diff
@@ -111,8 +111,10 @@ $(document).ready(function () {
         reset: function () {
             var defaults = this.model.defaults();
             this.model.set(defaults);
+            this.model.attributes.mode = this.device.attributes.features['--mode'].default; // Set colour/mode to device default
```

Comment:
Your suspicion that there's a slightly better way was correct.

I think, ultimately the best way would be to change `ScanRequest.defaults()` to use the Device (if it's there - it may not be on page load) - but the way the code is structured would make that a bit more painful right now.

In the mean time, if you do this:
```
defaults.mode = this.device.attributes.features['--mode'].default;
```
... just after we get `defaults` and before `this.model.set(defaults)` then Backbone `model.set` will take care of rendering for you and you don't need `this.render` any more.

Reply:
I agree that changing the ``defaults()`` would be the best option, but it seems like Backbone doesn't *really* support changing them after extending/creating the Model.

It seems to be possible tho to just override the ``defaults()`` function in ``deviceSync``:

```js
this.model.defaults = function() {
    return {
        // LocalStorage requires an id in order to
        // update / read. This can be fixed to anything
        id: 0,
        top: 0,
        left: 0,
        width: 215,
        height: 297,
        resolution: 150,
        mode: device.attributes.features['--mode'].default,
        brightness: 0,
        contrast: 0,
        convertFormat: 'tif',
        dynamicLineart: true,
        multiplePages: false
    };
};
```

But I'm even more unsure about this solution because it also doesn't seem to be very good.

Reply:
Yeah, I think I'm with you in terms of being unsure. I suspect the cleanest way might be to change within the `default()` itself but if doesn't have access to `this.device`.

I say, for the time being go with the:

```javascript
defaults.mode = this.device.attributes.features['--mode'].default;
```

At least it's contained to one line of code for the time being. If we need to do more later with other defaults then perhaps we revisit. What do you think?

Reply:
Yeah, I think for now that is the best solution.
Should I just do another commit and add it to the pull request (if that's even possible) or rather remove the current commit from history, make and commit the change, and open a new pull request?

Reply:
Just do another commit and push. (My personal view is not to remove commits unless something has gone really horribly wrong!). It should automatically update into this PR.